samples/net/mqtt/src/modules/transport/transport.c

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+#include <zephyr/smf.h>
+#include <net/mqtt_helper.h>
+
+#include "client_id.h"
+#if defined(CONFIG_MODEM_KEY_MGMT)
+#include "credentials_provision.h"
+#endif /* CONFIG_MODEM_KEY_MGMT */
+#include "message_channel.h"
+
+/* Register log module */
+LOG_MODULE_REGISTER(transport, CONFIG_MQTT_SAMPLE_TRANSPORT_LOG_LEVEL);
+
+/* Register subscriber */
+ZBUS_SUBSCRIBER_DEFINE(transport, CONFIG_MQTT_SAMPLE_TRANSPORT_MESSAGE_QUEUE_SIZE);
+
+/* ID for subscribe topic - Used to verify that a subscription succeeded in on_mqtt_suback(). */
+#define SUBSCRIBE_TOPIC_ID 2469
+
+/* Forward declarations */
+static const struct smf_state state[];
+static void connect_work_fn(struct k_work *work);
+
+/* Define connection work - Used to handle reconnection attempts to the MQTT broker */
+static K_WORK_DELAYABLE_DEFINE(connect_work, connect_work_fn);
+
+/* Define stack_area of application workqueue */
+K_THREAD_STACK_DEFINE(stack_area, 4096);
+
+/* Declare application workqueue. This workqueue is used to call mqtt_helper_connect(), and
+ * schedule reconnectionn attempts upon network loss or disconnection from MQTT.
+ */
+struct k_work_q queue;
```

Comment:
I think the queue should have a less generic name, easier to keep track of and when debugging.
Also should probably be `static`